### PR TITLE
use json-iterator, lint

### PR DESCRIPTION
**Pull request by s3rj1k**
Signed-off-by: s3rj1k <evasive.gyron@gmail.com>

### CODE REVIEW

**Comment by s3rj1k** [comment]
would be nice to see all of those PRs merged, so I could do mere test Relate PRs

getting hard to fix merge issues

**Comment by osamingo** [comment]
@s3rj1k I think this PR's nice challenge. But I don't feel this PR's benefit.
So I wanna see benchmarks and so on.

BTW, `osamingo/jsonrpc` version 3 is planning `Custom JSON Codec (Define Encoder interface)`.

File: unmarshal.go [hunk 1]
```diff
@@ -1,13 +1,13 @@
 package jsonrpc
 
-import "github.com/intel-go/fastjson"
+import json "github.com/json-iterator/go"
 
 // Unmarshal decodes JSON-RPC params.
-func Unmarshal(params *fastjson.RawMessage, dst interface{}) *Error {
-	if params == nil {
+func Unmarshal(params *json.RawMessage, dst interface{}) *Error {
+	if params == nil || dst == nil {
```
Comment (s3rj1k) [inline unmarshal.go]: @osamingo you should notice this change, not all codecs check for nil value

**Comment by s3rj1k** [comment]
@osamingo the main concern about github.com/intel-go/fastjson that it is not actively developed.
json-iterator on the other hand is actively developed, according to upstream does not have problems with interface fields and seem good replacement until v3 lib is released

**Comment by s3rj1k** [comment]
@osamingo also you can see benchmarks on main [json-iterator](https://github.com/json-iterator/go) github page

**Comment by osamingo** [comment]
@s3rj1k Thanks for your quick reply!

The `json-iterator` use `unsafe` package.
Google App Engine isn't approving use to `unsafe` and `syscall` packages.
If you wanna use `json-iterator` package, separate out logic with buildtag and `goapp` testing.

**Comment by osamingo** [comment]
This PR has been abandoned and has been closed.

**Pull request closed without merging**